src/main/java/com/github/biancacristina/compiler/syntax/grammar/builder/GrammarBuilder.java

```diff
@@ -10,17 +15,42 @@
 
     private GrammarBuilder(){}
 
-    public GrammarInterface build(ArrayList<ArrayList<String>> rules) {
+    public GrammarInterface build(ArrayList<RuleRequest> rules) {
         GrammarInterface grammar = new Grammar();
-        // TODO for each grammar rule, build its sentence and add on the grammar
+        rules.forEach( rule -> buildRule(grammar, rule.getProduction(), rule.getSentences()) );
         return grammar;
     }
 
-    private void buildSentence(GrammarInterface grammar, ArrayList<String> sentenceItems) {
-        // TODO add each sentenceItem on grammar
+    private void buildRule(GrammarInterface grammar, String productionLabel, String sentenceString) {
+        Sentence sentence = buildSentence(grammar, sentenceString);
+        Production production;
+        if (grammar.hasItem(productionLabel)) {
+            production = (Production) grammar.get(productionLabel);
+            production.addSentence(sentence);
+        } else {
+            production = new Production(productionLabel);
+            production.addSentence(sentence);
+            grammar.put(production);
+        }
+    }
+
+    private Sentence buildSentence(GrammarInterface grammar, String sentenceString) {
```

Comment:
Considering that a label can be a String (not just a char), shouldn't it be something like an Array of strings instead?
```suggestion
    private Sentence buildSentence(GrammarInterface grammar, ArrayList<String> sentenceStrings) {
```